### PR TITLE
Only enable JS runtime limits during execution

### LINKED ISSUE #5731: Handler invocations from js_kv_map_foreach get a full max_exec_time each
`js::Context::call()` sets a fresh start_time and max_exec_time in the runtime. Since this is called on each iteration by foreach, the total max duration is effectively reset every time. Found while looking at #5730.

### PR DESCRIPTION
**Pull request by achamayou**
Alternative to #5729 in addressing #5589. Also fixes #5731.

This change removes the interpreter heap, stack and runtime caps outside of JS code execution, which avoids having to handle those situations in the preparation of the runtime and of the response. In those contexts, the limits set by the environment (OS and/or SGX) apply instead, as they do in the rest of the code.

Note that this of course does not solve the issue of leaks in case a C++ exception is thrown in a wrapped function called from JS, this will need separate changes.

Also changed runtime options setting so that governance always has at least the default values available, to avoid getting to an unrecoverable situation where the values are so low that no further governance is possible.

### CODE REVIEW

File: src/js/wrap.cpp [hunk 1]
```diff
@@ -137,21 +137,40 @@ namespace ccf::js
...
     }
+    rt.set_runtime_options(tx);
     const auto curr_time = ccf::get_enclave_time();
     interrupt_data.start_time = curr_time;
     interrupt_data.max_execution_time = rt.get_max_exec_time();
-    interrupt_data.access = access;
     JS_SetInterruptHandler(rt, js_custom_interrupt_handler, &interrupt_data);
 
+    const auto rv = W(JS_Call(
+      ctx, f, ccf::js::constants::Undefined, argv.size(), argvn.data()));
```
Comment (eddyashton) [inline src/js/wrap.cpp:158]: Can this call `inner_call`?

**Review by eddyashton — APPROVED** [review]

File: src/js/wrap.cpp (hunk 1 above)
Comment (achamayou) [reply src/js/wrap.cpp]: Sure

File: tests/js-custom-authorization/custom_authorization.py [hunk 2]
```diff
@@ -111,6 +111,26 @@ def test_stack_size_limit(network, args):
         r = c.post("/app/recursive", body={"depth": safe_depth})
         assert r.status_code == http.HTTPStatus.OK, r
 
+        # Lower the cap until we likely run out of stack out of user code,
+        # and check that we don't crash and return an error message
```
Comment (eddyashton) [inline tests/js-custom-authorization/custom_authorization.py:115]:
```suggestion
        # and check that we don't crash. Check we return an error message
```

File: tests/js-custom-authorization/custom_authorization.py [hunk 3]
```diff
@@ -138,6 +158,27 @@ def test_heap_size_limit(network, args):
...
+        cap = safe_size
+        while cap > 0:
+            cap //= 2
+            LOG.info(f"Heap max size: {cap}")
+            with temporary_js_limits(network, primary, max_heap_bytes=cap):
+                r = c.post("/app/alloc", body={"size": 1})
+                if r.status_code == http.HTTPStatus.INTERNAL_SERVER_ERROR:
+                    message = r.body.json()["error"]["message"]
+                    assert message == "Exception thrown while executing.", message
+                    LOG.info(f"Out of memory at max heap size: {cap}")
```
Comment (eddyashton) [inline tests/js-custom-authorization/custom_authorization.py:172]: Should we be logging this discovered for the other cases? Or is this just a debug log that can be removed?

**Review by eddyashton — APPROVED** [review]

File: tests/js-custom-authorization/custom_authorization.py (hunk 3 above)
Comment (achamayou) [reply tests/js-custom-authorization/custom_authorization.py:172]: Just debug, it's not very precise anyway (it's a value at which we are out of memory, but not the highest).

Comment (achamayou) [inline tests/js-custom-authorization/custom_authorization.py:172]:
```suggestion
```

**Pull request merged**